registry/reference.go

```diff
@@ -40,6 +40,8 @@ var (
 	// The docker and OCI spec have the same regular expression.
 	// Reference: https://github.com/opencontainers/distribution-spec/blob/main/spec.md#pulling-manifests
 	tagRegexp = regexp.MustCompile(`^[\w][\w.-]{0,127}$`)
+
+	pathRegex = regexp.MustCompile(`^[a-z-][-\w.]*(:\w[\w.-]{0,127})?(@\w+:[0-9a-f]+)?$`)
```

Comment:
This can be tightened further, but it aims to be an improvement over the current validation.

Reply:
This has been reverted.